### PR TITLE
Support load from safe tensors

### PR DESCRIPTION
**Pull request by oovm**
Solve #53 

It can also work with official safetensors without convert weights now!

https://huggingface.co/runwayml/stable-diffusion-v1-5/blob/main/unet/diffusion_pytorch_model.safetensors

```sh
cargo run --example stable-diffusion --features clap -- --prompt "A rusty robot holding a fire torch." --sd-version=v1-5 --intermediary-images --unet-weights="data/diffusion_pytorch_model.safetensors"
```

Wrap function can be removed when https://github.com/LaurentMazare/tch-rs/pull/633 merged.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,4 @@ __pycache__
 .*~
 sd_*jpg
 sd_*png
+.idea/
```
Comment (vlad20012) [inline .gitignore:15]: You forgot the newline here :)

File: src/pipelines/stable_diffusion.rs [hunk 2]
```diff
@@ -150,7 +154,11 @@ impl StableDiffusionConfig {
         let mut vs_unet = nn::VarStore::new(device);
         let unet =
             unet_2d::UNet2DConditionModel::new(vs_unet.root(), in_channels, 4, self.unet.clone());
-        vs_unet.load(unet_weights)?;
+        if unet_weights.ends_with("safetensors") {
```
Comment (vlad20012) [inline src/pipelines/stable_diffusion.rs:157]: I suppose it's better to check the extensions with a dot:

```suggestion
        if unet_weights.ends_with(".safetensors") {
```

**Comment by vlad20012** [comment]
Also, with this PR it's now possible to load only unet weights, but not vue or clip weights. Is it expected? :thinking:

**Comment by LaurentMazare** [comment]
I don't think this PR is useful anymore now that supports for safetensors has been included in the tch crate ([tch#636](https://github.com/LaurentMazare/tch-rs/pull/636)) - loading from safetensors file should work for all models. So I'll close this PR but let me know if I'm missing something.

**Pull request closed without merging**